serialization/src/main/scala/sbt/serialization/json/JSONPickleFormat.scala

```diff
@@ -38,6 +36,15 @@ package json {
     /** The value in the pickled parsed into a JValue AST.  note this may throw. */
     def parsedValue: JValue
 
+    private[serialization] def readTypeTag: Option[String] = parsedValue match {
```

Comment:
combining this with the other similar method doesn't work out that well because there are several differences between the two. I think having a shared val for `"$type"` avoids the dangerous aspect of the duplication.